src/context/StationsContext.js

```diff
@@ -0,0 +1,58 @@
+import { createContext, useEffect, useState } from 'react';
+import Cookies from 'js-cookie';
+
+import { setBasicRadioStationsToLS } from '../utils/setBasicRadioStations';
+export const StationsContext = createContext();
+
+export const StationsContextProvider = ({ children }) => {
+  const [isModalOpen, setModalIsOpen] = useState(false);
+  const [radioStationsListLS, setRadioStationsListLS] = useState([]);
+
+  useEffect(() => {
+    if (!Cookies.get('visited')) {
+      Cookies.set('visited', true);
+      // set basic radio stations in local storage on first visit
+      setBasicRadioStationsToLS();
+    }
+    refreshRadioStations();
+  }, []);
+
+  function refreshRadioStations() {
+    let radioStations = JSON.parse(localStorage.getItem('stations'));
+    setRadioStationsListLS(radioStations);
+  }
+
+  function addNewRadioStationToLS(data, setModalIsOpen, radioStationsListLS) {
+    setModalIsOpen(false);
+
+    const stationFromForm = getRadioStationFromForm(data);
+    const stationsFromLocalStorage = radioStationsListLS;
+    stationsFromLocalStorage.push(stationFromForm);
+    localStorage.setItem('stations', JSON.stringify(stationsFromLocalStorage));
+    refreshRadioStations();
+  }
+
+  function getRadioStationFromForm(data) {
```

Comment:
Same as the above applies for this function here: your function doesn't have to care where the data comes from. It could come from the form, from an API call, or just another function or class that called it. This function could be names `parseRadioStationData`, `createRadioStationFromData`, or something like that